desktop-widgets/tab-widgets/TabDiveInformation.cpp

```diff
@@ -130,10 +147,35 @@ void TabDiveInformation::divesChanged(dive_trip *trip, const QVector<dive *> &di
 	case DiveField::WATER_TEMP:
 		ui->waterTemperatureText->setText(get_temperature_string(current_dive->watertemp, true));
 		break;
+	case DiveField::ATM_PRESS:
+		ui->atmPressVal->setText(ui->atmPressVal->text().sprintf("%d",current_dive->surface_pressure.mbar));
+		break;
 	case DiveField::DATETIME:
 		updateWhen();
 		break;
 	default:
 		break;
 	}
 }
+
+void TabDiveInformation::on_atmPressVal_editingFinished()
+{
+	int32_t atmpress;
+	if ((current_dive) && (!current_dive->userflags))
```

Comment:
Hmm - I may be misunderstanding the code... if the pressure was calculated from the dive computer(s), then you aren't able to overwrite it here. Is that what you intended?
Wouldn't it be more logical to not allow editing instead of silently ignoring the value the user entered? Or is this doing something else and I'm just confused?

Reply:
At the moment, if the airpressure was supplied by the dive computer, the edit box is non-editable and the combobox is disabled. I am heavily influenced by my interaction with the darktable software where the original photo is sacrosanct: you never change anything about the original photo and any editing is in xml outside the original. In the same way I am biased towards keeping the downloaded dive log as "pure" as possible. If you have other feelings I would be more than happy to fit in.

Reply:
> At the moment, if the airpressure was supplied by the dive computer, the edit box is non-editable and the combobox is disabled. I am heavily influenced by my interaction with the darktable software where the original photo is sacrosanct: you never change anything about the original photo and any editing is in xml outside the original.

I'm not sure that the analogy is fitting. After all, the whole point of darktable is to edit the image, just that the original is left untouched. Making the field editable without overwriting the original data would be the proper equivalent.

I see the use cases for editing the pressure makes sense: For example if you have a dive computer with a broken air-pressure sensor.

Reply:
A broken pressure sensor would render the dive computer unusable because it would not be able to read depth (well, more strictly, water pressure). But, Berthold I am totally open-minded about this. I would like to hear Dirk's or any one else's opinion. If one can over-write dive->surface_pressure even when dc->surface_pressure also exists, it would eliminate the need for any flag and it would simplify things. I used to have a facility that if one typed "0" into the airpressure text box and if dc->surface_pressure does exist, it puts dc->surface_pressure back into the text box. I removed this but can easily put it back. It was intended as mechanism for getting back the dc->surface_pressure after one has over-typed airpressure in the text box.

Reply:
Ok - I thought the altimeter was a different sensor, as this is a value that is notoriously hard to measure.

Another use case might be for those who use more than one dive computer and trust one over than the other. Or perhaps for people who care more about the geographic altitude than the actual pressure.

Reply:
Thanks for this, Berthold. Unless there are any other opinions, I will make dive->surface_pressure editable, even if dc->surface_pressure does exist. Would this be ok?